dot/state/pruner.go

```diff
@@ -0,0 +1,363 @@
+package state
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+)
+
+const (
+	journalPrefix = "journal"
+	lastPrunedKey = "last_pruned"
+	pruneInterval = time.Second
+)
+
+// nolint
+const (
+	FullNode    = "full"
+	ArchiveNode = "archive"
+)
+
+// Pruning online pruning mode of historical state tries
+type Pruning string
+
+// IsValid checks whether the pruning mode is valid
+func (p Pruning) IsValid() bool {
```

Comment:
```suggestion
func (p Pruning) Valid() bool {
```

Reply:
Function name indicates an action. The same practice is followed in Go standard library. https://github.com/golang/go/blob/962d5c997af450af1de9a38eb6510cdfc86ea689/src/encoding/json/decode.go#L362